packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeActionButton.tsx

```diff
@@ -24,26 +30,34 @@ export type FilterActionProps = {
   /** Action to perform when the filter button is clicked for this node. */
   onFilter?: (hierarchyLevelDetails: HierarchyLevelDetails) => void;
 
-  label: string;
+  label?: string;
 } & Partial<Pick<ReturnType<typeof useTree>, "getHierarchyLevelDetails">>;
 
 /** @internal */
 export function TreeActionButton({ show, label, action, icon }: TreeItemAction) {
-  return (
-    <IconButton className={`tree-action-item${!show ? "-invisible" : ""}`} variant={"ghost"} onClick={action} label={label} icon={icon ?? placeholderIcon} />
-  );
+  return <Tree.ItemAction visible={show} onClick={action} label={label} icon={icon ?? placeholderIcon} />;
 }
 
 /** @alpha */
 export function createFilterAction({ onFilter, getHierarchyLevelDetails, label }: FilterActionProps): (node: PresentationHierarchyNode) => TreeItemAction {
   return (node) => {
+    const handleVisibility = () => {
+      if (!onFilter || !node.isFilterable) {
+        return false;
+      }
+      if (node.isFiltered) {
+        return true;
+      }
+      return undefined;
+    };
+
     return {
-      label,
+      label: label ?? defaultLocalizedStrings.filterHierarchyLevel,
```

Comment:
`defaultLocalizedStrings` should not be used here. I think this could be handled in `TreeActionButton` by checking if `label` mathes any of the localized strings that we support.

Reply:
Changed